### PR TITLE
SDIT-1748 update a sentence

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt [hunk 1]
```diff
@@ -532,6 +532,38 @@ class SentencingService(
...
+        telemetryClient.trackEvent(
+          "sentence-updated",
+          mapOf(
+            "bookingId" to bookingId.toString(),
+            "sentenceSequence" to sentenceSequence.toString(),
+          ),
+          null,
+        )
+      }
+    }
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt:564]: shouldn't you be throwing NotFoundExceptions in both these case ?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt [hunk 2]
```diff
@@ -2987,7 +2988,238 @@ class SentencingResourceIntTest : IntegrationTestBase() {
     }
 
     @AfterEach
-    internal fun deletePrisoner() {
+    internal fun deleteSentence() {
+      repository.deleteSentenceByBookingId(latestBookingId)
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt:2992]: as a matter of interest if you just delete the offender won't it cascade down to case and charge?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt (hunk 2 above)
Comment (lbennett-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt:2992]: no it doesn't - I could look into why

**Pull request merged**